plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java

```diff
@@ -617,9 +633,37 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
             }
             newDomain = TupleDomain.withColumnDomains(supported);
             remainingFilter = TupleDomain.withColumnDomains(unsupported);
+
+            if (isComplexExpressionPushdown(session)) {
+                ImmutableList.Builder<String> newExpressions = ImmutableList.builder();
+                ImmutableList.Builder<ConnectorExpression> remainingExpressions = ImmutableList.builder();
+                for (ConnectorExpression expression : extractConjuncts(constraint.getExpression())) {
```

Comment:
Why do we need to extract the Conjuncts ? Why can't we have a single String which could represent or partially represent the filter expression ?